runtime/static_config.go

```diff
@@ -389,10 +390,32 @@ func (conf *StaticConfig) parseFile(
 		))
 	}
 
-	var object map[string]interface{}
+	var object map[interface{}]interface{}
 	err := yaml.Unmarshal(bytes, &object)
 	if err != nil {
 		panic(err)
 	}
-	return object
+	return stringizeMapKey(object).(map[string]interface{})
```

Comment:
Won't this type assertion fail if you take the `[]interface{}` case in `stringizeMapKey`? Since it passes tests, does that make the `[]interface{}` case dead code if doesn't occur?

Reply:
`object` is declared of type `map[interface{}]interface{}`, so the conversion should not fail. The `[]interface{}` case is for when the value is a slice of maps where the keys are also interface{} and need to be stringized.